src/shim/runtime/v2/rune/v2/create_test.go

```diff
@@ -0,0 +1,141 @@
+// The codebase is inherited from kata-containers with the modifications.
+
+package v2
+
+import (
+	"context"
+	_ "embed"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/confidential-containers/enclave-cc/src/shim/runtime/v2/rune/config"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/runtime/v2/runc"
+	taskAPI "github.com/containerd/containerd/runtime/v2/task"
+	"github.com/prashantv/gostub"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	testSandboxID   = "777-77-77777777"
+	testContainerID = "42"
+	testagentID     = "43"
+	testImage       = "docker.io/library/busybox:latest"
+
+	testContainerTypeAnnotation    = "io.kubernetes.cri.container-type"
+	testSandboxIDAnnotation        = "io.kubernetes.cri.sandbox-id"
+	testSandboxNamespaceAnnotation = "io.kubernetes.cri.sandbox-namespace"
+	testImageAnnotation            = "io.kubernetes.cri.image-name"
+	testContainerTypeSandbox       = "sandbox"
+	testContainerTypeContainer     = "container"
+	testNamespaceTypeDefault       = "default"
+	testNamespaceTypeKubesystem    = "kube-system"
+)
+
+const (
+	testDirMode  = os.FileMode(0750)
+	testFileMode = os.FileMode(0640)
+)
+
+//go:embed testdata/busybox.json
+var busyboxConfigJson []byte
+
+func SetupOCIConfigFile(t *testing.T) (rootPath string, bundlePath, ociConfigFile string) {
+	assert := assert.New(t)
+
+	tmpdir := t.TempDir()
+
+	bundlePath = filepath.Join(tmpdir, "bundle")
+	err := os.MkdirAll(bundlePath, testDirMode)
+	assert.NoError(err)
+
+	ociConfigFile = filepath.Join(bundlePath, "config.json")
+	err = os.WriteFile(ociConfigFile, busyboxConfigJson, testFileMode)
+	assert.NoError(err)
+
+	return tmpdir, bundlePath, ociConfigFile
+}
+
+func TestCreatePod(t *testing.T) {
```

Comment:
I think we'll need to specify built-in defaults and don't fail on missing files:
```
    create_test.go:95: 
        	Error Trace:	create_test.go:95
        	Error:      	"open /etc/enclave-cc/config.toml: no such file or directory" does not contain "OCI runtime create failed"
        	Test:       	TestCreatePod
```

Reply:
hi. In shim code, it will detect whether pre-installed OCI bundles (agent enclave instance and boot instance) and shim configure existed. If not, it will print the error. 

So if the developer and user want to test shim, they need to have pre-installed OCI bundles (agent enclave instance and boot instance) and shim configure first.
```
--- FAIL: TestCreatePod (0.00s)
    create_test.go:95: 
                Error Trace:    create_test.go:95
                Error:          "pre-installed OCI bundle should provided in /opt/enclave-cc/agent-instance/" does not contain "OCI runtime create failed"
```

Reply:
> So if the developer and user want to test shim, they need to have pre-installed OCI bundles

I'm expecting that we setup our Github Actions CI to run these unit tests for every subsequent PR in this repository. That is why we need to make these tests to just run `go test -v ./...` without any dependencies to what is pre-installed. This case is similar to what we had with `/run/enclave-cc/`:  let's just have `<tmpdir>/opt/enclave-cc/agent-instance/` etc. created before running the tests.